app/controllers/LoginController.scala

```diff
@@ -47,7 +47,12 @@ object LoginController {
         val validLength = cleanRefNumber.length > 9 && cleanRefNumber.length < 12: Boolean
         validLength
       }),
-      "postcode" -> nonEmptyTextOr("postcode", loginPostcode),
+      "postcode" -> text.verifying(Errors.invalidPostcodeOnLetter, pc => {
+        var cleanPostcode = pc.replaceAll("[^\\w\\d]", "")
```

Comment:
It might be worth, adding a comment to describe what is the intent, as it is not straight forward